src/components/vm/hostdevs/hostDevCard.jsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2021 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import cockpit from 'cockpit';
+import { vmId, findHostNodeDevice } from "../../../helpers.js";
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+/* Adds an optional description-value pair to an array which represents multiple values of a table cell
+ *
+ * @param chunks an array into which element is added to
+ * @param value a value of the descr-value pair
+ * @param descr a description of the descr-value pair
+ * @param id
+ */
+function addOptionalToCell(chunks, value, id, descr) {
```

Comment:
I'm not sure. It does not render any component. It just a helper function which ADDS a component to a list. That list is rendered somewhere else, no?
So I think calling such function which only adds to a list but doesn't render anything with a react functional component interface (i.e. `<AddOptionalCell chunks={chunks} ... />`) would be confusing.

Reply:
A React component does not have to be a whole part of the page, it can definitely be a row in a table, or a row in some other structure like here. 

Basically we are just discussing if returning React Node types makes sense to do in normal functions or we should write them as React functional components. See https://stackoverflow.com/questions/64453528/in-react-what-is-the-difference-between-using-functional-component-and-just-a-f

In any case, you need to decide, either you use a normal function and don't use propTypes, or a functional component with proptypes.